operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/ManagedInformerEventSource.java

```diff
@@ -39,4 +40,8 @@ public void stop() {
     super.stop();
     manager().stop();
   }
+
+  public void populateCacheUpdatedResource(R resource) {
+    cache.put(ResourceID.fromResource(resource), resource);
```

Comment:
Based on https://github.com/fabric8io/kubernetes-client/pull/3654#issuecomment-1026838969, I'm not sure we should be modifying the informer's cache :(
Another thing is that this won't be possible in 6.0 anymore so we need to think about an alternative, in particular in the light of: https://github.com/fabric8io/kubernetes-client/pull/3616. So we might have to think about re-adding our own caching layer sooner than later…

Reply:
Yes, the plan is to do it with the separate cache if the informers will be modified for that. Not sure if we can get away with this until that time. It's not 100% correct, but the chance to have there a race condition is very close to zero

Reply:
An alternative is to wait for the event that the cache received the resource with the resource version.

Reply:
That would mean not the update/delete would propagate the cache but the `getResource()` will be kinda "synchronous"

Reply:
Solved it in a different way, pls see updated PR. (Will add unit tests soon)